src/components/masonry_grid.tsx

```diff
@@ -0,0 +1,137 @@
+import { times } from "lodash"
+import PropTypes from "prop-types"
+import * as React from "react"
+
+interface Props extends React.HTMLProps<MasonryGrid> {
+  columnCount?: number
+  columnMargin?: number
+  getDisplayComponent: (artwork: object) => any
+  getAspectRatio: (artwork: object) => number
+  items: Array<object>
+  rowMargin?: number
+}
+
+export default class MasonryGrid extends React.Component<Props, void> {
+  static propTypes = {
+    columnCount: PropTypes.number,
+    columnMargin: PropTypes.number,
+    getDisplayComponent: PropTypes.func.isRequired,
+    getAspectRatio: PropTypes.func.isRequired,
+    items: PropTypes.array.isRequired,
+    rowMargin: PropTypes.number,
+  }
```

Comment:
Ah yeah, runtime assertions will be required again for safety when integrating into a vanilla JS code base.